AMW_angular/io/src/app/settings/environments/environments-page.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, inject, OnInit } from '@angular/core';
+import { IconComponent } from '../../shared/icon/icon.component';
+import { EnvironmentService } from '../../deployment/environment.service';
+import { ToastService } from '../../shared/elements/toast/toast.service';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { Environment, EnvironmentTree } from '../../deployment/environment';
+import { takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-environments-page',
+  standalone: true,
+  imports: [IconComponent],
+  templateUrl: './environments-page.component.html',
+})
+export class EnvironmentsPageComponent implements OnInit {
+  private environmentsService = inject(EnvironmentService);
+  private toastService = inject(ToastService);
+  private error$ = new BehaviorSubject<string>('');
+  private destroy$ = new Subject<void>();
+  environmentTree: EnvironmentTree[];
+
+  ngOnInit(): void {
+    this.error$.pipe(takeUntil(this.destroy$)).subscribe((msg) => {
+      msg !== '' ? this.toastService.error(msg) : null;
+    });
```

Comment:
I think you don't need to handle errors here. Errors from http requests are already handled globally.

And If you need to show an error from within your component, just use the toastservice (or maybe throw an error)